keystore/v2/keystore/keyStore.go

```diff
@@ -71,6 +92,81 @@ func NewTranslatorKeyStore(keyStore api.MutableKeyStore) *TranslatorKeyStore {
 	}
 }
 
+// ListKeys enumerates keys present in the key store.
+func (s *ServerKeyStore) ListKeys() ([]keystore.KeyDescription, error) {
+	keyRings, err := s.ListKeyRings()
+	if err != nil {
+		return nil, err
+	}
+	keys := make([]keystore.KeyDescription, len(keyRings))
+	for i := range keys {
+		description, err := s.describeKeyRing(keyRings[i])
+		if err != nil {
+			return nil, err
+		}
+		keys[i] = *description
+	}
+	return keys, nil
+}
+
+func (s *ServerKeyStore) describeKeyRing(path string) (*keystore.KeyDescription, error) {
+	if path == poisonKeyPath {
+		return &keystore.KeyDescription{
+			ID:      path,
+			Purpose: PurposePoisonRecord,
+		}, nil
+	}
+	if path == authKeyPath {
+		return &keystore.KeyDescription{
+			ID:      path,
+			Purpose: PurposeAuthentication,
+		}, nil
+	}
+
+	components := strings.Split(path, string(filepath.Separator))
```

Comment:
can you add comment with example of path for each component? like '/<client_id>/<something_else>/<storage>'? because looking on it hard to understand which strings expected here
plus I forgot how we store our keys. Is it same tree structure of files in keystore v2 or it one file in asn1 format which contains binary tree structure inside?

Reply:
> Is it same tree structure of files in keystore v2 or it one file in asn1 format which contains binary tree structure inside?

Each key ring is stored in an individual ASN.1 file. Directory structure mirrors the IDs, so if you have `client/lagovas/storage` then this keyring is stored in a file `${keystore}/client/lagovas/storage.keyring`.